### DISCUSSION TITLE
Ağaç Oyunu Kolay Versiyon

### DISCUSSION THREAD

**Discussion by Tunapotext** [sonsuzus/sonsuzus#477] [Soru Cevap]
---

Problem Tanımı (Kolay Versiyon):

Bu versiyonun farkı, `t=1` olması ve bir dizi gibi görünen bir ağaç üzerinde çalışmamızdır.

Ron ve Hermione, başlangıçta aktif olmayan n düğümden oluşan bir ağacın üzerinde oyun oynuyorlar. Bu ağaç özel bir ağaçtır çünkü tam olarak iki yaprağa sahiptir. Bu nedenle, bu ağaç bir dizi gibi görülebilir. Oyun, her biri taşın başlangıçta tam olarak bir düğümde olduğu ve bu düğümün aktif olduğu `t` turdan oluşur. Bir hamle, taşın bulunduğu düğümün bir komşusuna taşınması ve bu komşunun aktif hale gelmesi şeklindedir. Ron ilk hamleyi yapar, ardından sırayla Hermione oynar ve geçerli bir hamle yapılmadığı zaman oyunun kaybedeni belirlenir. Her iki oyuncu da optimizasyon yaparak oynarsa, her turda kim kazanır?

Not: Tüm turlar aynı ağaçta oynanır; sadece başlangıç düğümü değişir. Ayrıca, her turdan sonra tüm aktif düğümler tekrar inaktif hale gelir.

Girdi:
İlk satırda, ağaçtaki düğüm sayısını (n) ve tur sayısını (t=1) belirten iki tam sayı bulunur.

Sonraki `n-1` satırda, ağaçtaki her bir kenarı temsil eden iki tam sayı 1≤u,v≤n bulunur.

Son satırda, taşın başlangıçta konduğu düğümün numarasını belirten t tam sayısı bulunur.

Çıktı:
Her test için, sonucu belirten bir satır olacaktır: "Ron" veya "Hermione".

---

Örnekler:

Örnek 1:

Girdi:
```
3 1
2 3
3 1
3
```

Çıktı:
```
Ron
```

Örnek 2:

Girdi:
```
5 1
1 2
2 3
3 4
4 5
5
```

Çıktı:
```
Hermione
```
def solve():
    n, t = map(int, input().split())
    
    a = [[] for _ in range(n + 1)]  # Komşuluk listesi
    for _ in range(n - 1):
        x, y = map(int, input().split())
        a[x].append(y)
        a[y].append(x)
    
    s = int(input())  # Başlangıç düğümü
    l = []
    
    for i in range(1, n + 1):
        if len(a[i]) == 1:
            l.append(i)
    
    if s == l[0] or s == l[1]:
        print("Ron")
        return
    
    d1 = 0
    cur = s
    while cur != l[0]:
        for i in a[cur]:
            if i != cur:
                cur = i
                d1 += 1
    
    d2 = 0
    cur = s
    while cur != l[1]:
        for i in a[cur]:
            if i != cur:
                cur = i
                d2 += 1
    
    if min(d1, d2) % 2 == 0:
        print("Hermione")
    else:
        print("Ron")

**Comment by sonsuzus** [comment]
Birincisi cevabı yorum olarak yazmalısın.

İkincisi yapay zeka cevaplarını kabul etmiyoruz.
